### ISSUE TITLE
macOS Monterey: Triggers warning because outdated python is called

### ISSUE THREAD

**Issue by datenreisender** [sebhildebrandt/systeminformation#627]
**Describe the bug**
When `versions()` is called on macOS in desktop apps, a warning is triggered, because the outdated python 2 is called.

**To Reproduce**
The easiest way to reproduce this is to create a minimal desktop application like I did with Electron in https://github.com/datenreisender/systeminformation-electron-test. If you clone that and run it with `npm start` on macOS Monterey, this windows is shown:

![image](https://user-images.githubusercontent.com/260705/142228635-e7fad4d0-4d5e-467b-845f-9c4f4479a7d3.png)

Even if the comment says that “Electron” needs to be updated, there is nothing about Electron that is outdated here. If the call to `systeminformation.versions()` is removed, then the warning is not shown. If I replace the invocation of `python` in https://github.com/sebhildebrandt/systeminformation/blob/148cb415999f07b7fe112d13c67acd329c1395c8/lib/osinfo.js#L785 with an invocation of `python3` (which of course then shows the wrong version) the warning also is not shown.

The `Learn More…` link points to https://www.python.org/doc/sunset-python-2/.

Please note, that macOS remembers if it has already shown you the warning. To circumvent that, you can simply rename the directory where the app or your sources are stored. After that renaming, the warning is again shown on the first invocation.

**Environment:**
 - systeminformation package version: 5.9.13
 - electron version: 16.0.0
 - OS: macOS Monterey 12.0.1

**Ideas to fix this:**
On macOS `/usr/bin/python` should not be called by default anymore. Some ideas how to fix this:
- Do the call only if a certain flag is passed to `systeminformation.versions()`. 
- Only when the property `python` is read, then the call is made. 
- Do not fill the property `python` on macOS at all anymore.
- Fill the property `python` on macOS only if something like `/usr/local/bin/python` is used, but never by calling `/usr/bin/python`.

**Comment by sebhildebrandt** [comment]
@datenreisender ... interesting, I have a completely new MacBook, with Monterey installed. Just a few days old. When I am calling `si.versions()` I get no warning, and all python versions are shown correctly. 

Ok, I installed Xcode command line tools ...  and `which python` gives back `/usr/bin/python` ... so I need to think about what the difference might be ... and how to avoid this error ...

**Comment by datenreisender** [comment]
Strange, I also have the Xcode command line tools installed. Do you maybe also have a version of python installed through homebrew? What is the output if you just call `/usr/bin/python` in a shell? I get this:
```

WARNING: Python 2.7 is not recommended.
This version is included in macOS for compatibility with legacy software.
Future versions of macOS will not include Python 2.7.
Instead, it is recommended that you transition to using 'python3' from within Terminal.

Python 2.7.18 (default, Oct  2 2021, 04:20:39)
[GCC Apple LLVM 13.0.0 (clang-1300.0.29.1) [+internal-os, ptrauth-isa=deploymen on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>>
```

**Comment by sebhildebrandt** [comment]
@datenreisender i need to check it later this evening. I have also one more new machine with no Xcode command line tools installed. Will try to find a solution. But I am sure that I will need you then to double check my approach ;-)

**Comment by sebhildebrandt** [comment]
@datenreisender so the current situation here:

- macOS Monterey 12.0.1
- Xcode command line tools installed (path '/Library/Developer/CommandLineTools/usr/bin/' exists)
- no python installed with homebrew
- `which python` returns `/usr/bin/python`
- `python -V` returns `Python 2.7.18` (nothing else, no warnings , ...)

Can you check the same info set on your machine?

**Comment by sebhildebrandt** [comment]
@datenreisender ... so what I see in the Changelog Monterey 12.0.1:

## Python
### Deprecations
If an app uses Python 2.7, macOS now triggers an alert indicating that the developer must update the app to ensure it will work in future versions of macOS. (80221011)

Seems that I did not see the warning? 

So I now created a fix for it ... code only pushed to GitHub. can you pull it from here and test it?? Should not call `python`(2) beginning with Mac OS 12.0.1 (except there is something installed with homebrew, then this is called)

If this fix works on your side, I will publish a new npm version.

**Comment by sebhildebrandt** [comment]
@datenreisender new version 5.9.14 just published.

One thing ... I was only calling `python -V` in my package, you are now reporting, that for this there is no warning ... maybe I am confused now?

**Comment by datenreisender** [comment]
> @datenreisender new version 5.9.14 just published.

🙏 Thanks

> One thing ... I was only calling `python -V` in my package, you are now reporting, that for this there is no warning ... maybe I am confused now?

Yes, that is really confusing. I was talking about two different kinds of warnings: The “… needs to be updated” dialog I showed above or a warning `WARNING: Python 2.7 is not recommended. […]` in the terminal window itself (that is output on STDERR).

When calling `python` (without any switch) then the warning `WARNING: Python 2.7 is not recommended. […]` is printed to STDERR in any terminal.

When calling `python -V` in Terminal.app or iTerm.app I also get no warning (neither on STDERR nor the dialog) but I have no clue whether that does not happen because Apple added a special exception for these terminal apps.

_But_ when I call `python -V` from a different terminal app that I never used before (e.g. [kitty](https://sw.kovidgoyal.net/kitty/)) _then_ I get the warning dialog. On every subsequent call, macOS seems to remember that it already warned about kitty and does not warn again. Of course, the “never used before” makes it harder to reproduce. What I found as a workaround: macOS seems to remember the apps it warned about by name. So, e.g. if I rename “kitty“ to “kitty - test”, then I get a warning on the first time I call `python -V` again:
![image](https://user-images.githubusercontent.com/260705/142580270-1ba0ea4a-393a-4c72-ac76-95f6c6e2b93b.png)
